app/scripts/app-init.js

```diff
@@ -164,3 +164,25 @@ const registerInPageContentScript = async () => {
 };
 
 registerInPageContentScript();
+
+/**
+ * creates an offscreen document that can be used to load iframes containing
+ * scripts that can communicate with the extension through the chrome runtime
+ * API. Only one offscreen document may exist, so iframes are used within the
+ * created offscreen document. See the offscreen folder for more details.
+ */
+async function createOffscreen() {
+  if (await chrome.offscreen.hasDocument()) {
+    return;
+  }
+
+  await chrome.offscreen.createDocument({
+    url: './offscreen/offscreen.html',
+    reasons: ['IFRAME_SCRIPTING'],
+    justification: 'Load HW scripts',
```

Comment:
From my understanding, this _may_ be shown to users. We should make clear that the offscreen will be used for hardware wallets and Snaps.

Reply:
When you say may be shown to users you're meaning the 'justification' string?

Reply:
Yeah.